ui/src/app/subnets/views/FabricDetails/FabricDetailsHeader/FabricDeleteForm/FabricDeleteForm.tsx

```diff
@@ -0,0 +1,71 @@
+import { useCallback } from "react";
+
+import { Notification } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+
+import FormikForm from "app/base/components/FormikForm";
+import type { EmptyObject } from "app/base/types";
+import { actions as fabricActions } from "app/store/fabric";
+import fabricSelectors from "app/store/fabric/selectors";
+import type { Fabric, FabricMeta } from "app/store/fabric/types";
+import type { RootState } from "app/store/root/types";
+import subnetSelectors from "app/store/subnet/selectors";
+import subnetURLs from "app/subnets/urls";
+import { isId } from "app/utils";
+
+type Props = {
+  closeForm: () => void;
+  id?: Fabric[FabricMeta.PK] | null;
+};
+
+const FabricDeleteForm = ({ closeForm, id }: Props): JSX.Element | null => {
+  const dispatch = useDispatch();
+  const fabric = useSelector((state: RootState) =>
+    fabricSelectors.getById(state, id)
+  );
+  const subnetsInFabric = useSelector((state: RootState) =>
+    subnetSelectors.getByFabric(state, id)
+  );
+  const errors = useSelector(fabricSelectors.errors);
+  const saved = useSelector(fabricSelectors.saved);
+  const saving = useSelector(fabricSelectors.saving);
+  const cleanup = useCallback(() => fabricActions.cleanup(), []);
+
+  if (!isId(id) || !fabric) {
+    return null;
+  }
+
+  const canBeDeleted = subnetsInFabric.length === 0;
+  return (
+    <FormikForm<EmptyObject>
+      buttonsBordered={false}
+      cleanup={cleanup}
+      errors={errors}
+      initialValues={{}}
+      onCancel={closeForm}
+      onSubmit={() => {
+        dispatch(cleanup());
+        dispatch(fabricActions.delete(id));
+      }}
+      savedRedirect={subnetURLs.indexBy({ by: "fabric" })}
```

Comment:
Just noticed this won't work as you expect. Redirect component used in FormikForm will skip the URL params, we'd need to pass this as a location descriptor object. Interestingly enough, `history.replace` works just fine with a path with params.

Reply:
Interestingly, I just tried to reproduce this and it was redirecting fine (even if I changed it to `by: "space"` it goes to `/r/networks?by=space`).

Reply:
Can also confirm it works on latest Firefox and Chrome